pandas/indexes/base.py

```diff
@@ -3478,6 +3478,15 @@ def _evaluate_numeric_binop(self, other):
             cls.__rdiv__ = _make_evaluate_binop(
                 operator.div, '__div__', reversed=True)
 
+        def divmod_constructor(result, **attrs):
+            return Index(result[0], **attrs), Index(result[1], **attrs)
+
```

Comment:
I think put the function in-line (as a lambda), since they are grouped might be more clear (but if not, then go with what you have)